utils/hre-extensions.ts

```diff
@@ -159,9 +170,27 @@ extendEnvironment((hre) => {
     },
   }
 
-  hre.getNamedSigner = async (name: string): Promise<Signer> => {
-    const accounts = await getNamedAccounts()
-    return ethers.provider.getSigner(accounts[name])
+  if (network.name == 'hardhat' || network.name == 'localhost') {
+    const getNamedAccountsOriginal = hre.getNamedAccounts
+    hre.getNamedAccounts = async () => {
+      const accounts = await getNamedAccountsOriginal()
+      accounts.deployer = '0xAFe87013dc96edE1E116a288D80FcaA0eFFE5fe5'
+      return accounts
+    }
+
+    hre.getNamedSigner = async (name: string): Promise<Signer> => {
+      if (name == 'deployer') {
+        await network.provider.request({
+          method: 'hardhat_impersonateAccount',
+          params: ['0xAFe87013dc96edE1E116a288D80FcaA0eFFE5fe5'],
+        })
+        return ethers.provider.getSigner(
+          '0xAFe87013dc96edE1E116a288D80FcaA0eFFE5fe5'
+        )
+      }
+      const accounts = await hre.getNamedAccounts()
+      return ethers.provider.getSigner(accounts[name])
+    }
```

Comment:
I believe we should be able to get rid of this change as well honestly with the update to the `hardhat.config.ts`